docs/service-overview.md

```diff
@@ -0,0 +1,947 @@
+them. By default, it is assumed that all the customer-owned serials are added to
+the root group. Users with the **ASSIGNER** or **ADMIN** role can move/add
+serials within the tree from there. A serial can be assigned to a single group
+apart from the root group. See the `/GetSerial` RPC for an example of the
+details available for a given serial number.
+
+## Pinned Domain Certificates (PDCs)
+
+Pinned Domain Certificates (aka PDCs) are the roots of trust used in the
+ownership voucher, along with “revocation checks” and an “expiry time”. A
+certificate ID is returned when a pinned domain cert is added to a group. While
+creating ownership vouchers, this cert ID is used to reference the certificate.
+See the `/GetDomainCert` RPC for an example of this data.
+
+## Roles
+
+Users are assigned to **ADMIN** or **REQUSTER** roles. An **ADMIN** can do
+everything that a (**ASSIGNER**, **REQUESTOR**) can do, along with requesting
+Ownership Vouchers. See `/GetGroup` RPC for an example. It is a subfield of
+group data.
+
+A vendor may also maintain specific internal roles for performing
+support-related operations. An **ADMIN** role can set up the group hierarchy and
+add users with different roles for different subtrees. An **ASSIGNER** can
+create PDCs and associate them with groups, move serials between groups, and
+issue ownership vouchers. A **REQUESTOR** role allows read access and the
+ability to issue ownership vouchers.
+
+<table>
+  <tr>
+   <td><strong>Role</strong>
+   </td>
+   <td><strong>Allowed RPCs</strong>
+   </td>
+   <td><strong>Allowed to be Assigned / Removed by</strong>
+   </td>
+   <td><strong>Allowed to be Assigned on</strong>
+   </td>
+   <td><strong>Allowed to be Viewed by</strong>
+   </td>
+  </tr>
+  <tr>
+   <td>SUPPORT
+   </td>
+   <td>All RPCs
+   </td>
+   <td>SUPPORT on the group/org or its parent
+   </td>
+   <td>the group/org or its children
+   </td>
+   <td>Anyone
+   </td>
+  </tr>
+  <tr>
+   <td>ADMIN
+   </td>
+   <td>All RPCs
+   </td>
+   <td>SUPPORT / ADMIN on the group/org or its parent
+   </td>
+   <td>the group/org or its children
+   </td>
+   <td>Anyone
+   </td>
+  </tr>
+  <tr>
+   <td>ASSIGNER
+   </td>
+   <td>All except CreateGroup, DeleteGroup, AddUserRole, RemoveUserRole,
+   </td>
+   <td>SUPPORT / ADMIN on the group/org or its parent
+   </td>
+   <td>the group/org or its children
+   </td>
+   <td>Anyone
+   </td>
+  </tr>
+  <tr>
+   <td>REQUESTOR
+   </td>
+   <td>GetGroup, GetUserRole, GetDomainCert, /≥≤, GetOwnershipVoucher
+   </td>
+   <td>SUPPORT / ADMIN / ASSIGNER on the group/org or its parent
+   </td>
+   <td>the group/org or its children
+   </td>
+   <td>Anyone
+   </td>
+  </tr>
+</table>
+
+## Groups
+
+Groups are used to limit the scope of available data and allow for the
+members of the group to manage a subset of items. It is assumed that individual
+vendors have different naming conventions for groups vs. the base organization.
```

Comment:
Could you please explain what assumed that individual
vendors have different naming conventions for groups vs. the base organization means?

I think the name of the root group might be a part of the service interface i.e. it must be  = org_id. The ids of any child groups can be anything and it is upto the vendor's implementation to name them however they wish.

Reply:
Looks good except for this bit. I'm uncertain what the above statement exactly implies, but it should be okay to merge this and we can add this when modifying the examples to make use of the component type